### DISCUSSION TITLE
깐부스터디#1주차 회고

### DISCUSSION THREAD

**Discussion by codesquadIOS** [codesquadIOS/OOP#2] [Announcements]
## 😎 싱하횽의 피드백
- 변하지 않는 하위데이터는 struct 로 하는 게 좋다(값 복사)
    - 함수내부에서 인자로 받는 값이 class 면, 참조로 넘겨 받는다.
- 테스트를 하면서 하자
    - TDD
- 구조를 처음부터 너무 크게 잡지 말자.
    - 나무 -> 나무 -> 나무 = 숲
    - 바리스타가 어떤 처리를 할 지 명확하게 알 수 있어서 좋다
    - 앞부분이 제대로 개발해야 뒷부분을 개발할 수 있다
    - 앞부분:
        - 손님이 주문을 어떻게 넣을지 생각을 하고 그 이후 흐름을 생각
        - ⇒ 그래서 테스트를 생각하면서 하는게 좋다

## 댓글로 회고를 남겨주세요
- 깐부 스터디를 하며 좋았던 점, 잘했던 점
- 아쉬웠던 점
- 다음주 스터디에 할 도전

**Comment by TaeKyeongKim** [comment]
상위 모듈에 대해 좀더 구체적으로 알수 있었고, 코드를 처음 짤때 어떤식으로 문제를 바라보고 시작해야할지 팁을 많이 받아서 좋았습니다 :) 
다음주는 조금더 적극적으로 네비게이터 역할에 참여해 보고 싶어요 ! (계속 하다보면 적응돼서 잘하겠쥬?)

**Comment by Jinsujin** [comment]
### 깐부 스터디를 하며 좋았던 점, 잘했던 점
- 같은 기능이지만, 이를 바라보는 다양한 생각들을 공유할 수 있어서 좋았습니다!
- 이런 스터디를 한다는 것 자체가 큰 의미가 있다고 생각하기 때문에, 같이 참여해 주신 분들에게 감사합니다!
  - 그동안 했던 스터디 중에서 가장 의욕이 넘치는 멤버들 🥰

### 아쉬웠던 점
- 중간부터 마음이 급해져 의견을 나누는 것 보다는 코드에 집중을 해서 아쉬웠습니다.

### 다음주 스터디에 할 도전
- 급하게 진행하지 말고 테스트 코드도 써보며 올바르게 짰는지 확인하는 과정을 거쳐야 겠습니다.
